guides/release/components/built-in-components.md

```diff
@@ -1,190 +1,215 @@
+  maxlength="50"
+/>
 ```
 
-Will bind the `disabled` attribute to the value of `entryNotAllowed` in the
-current context.
+Recall that there were a few exceptions. The following input attributes must be passed as arguments (i.e. do prepend `@`) to the `<Input>` component:
+
+- `@checked`
+- `@id`
+- `@type`
+- `@value`
+
 
 ### Actions
 
-To dispatch an action on specific events such as `key-down`, use the following
+Starting with Ember Octane, we recommend using the `{{on}}` modifier to call an action on specific events such as the [input event](https://developer.mozilla.org/docs/Web/API/HTMLElement/input_event).
 
 ```handlebars
 <label for="input-name">Name:</label>
-<Input id="input-name" name="name" @type="text" @value={{this.name}} @key-down={{this.updateName}} />
+<Input
+  @id="input-name"
+  @value={{this.name}}
+  {{on "input" this.validateName}}
+/>
 ```
 
-The following event types are supported (dasherized format):
+The event name (e.g. `"focusout"`, `"input"`, `"keydown"`) always follows the casing that the HTML standard uses.
 
-* `@enter`
-* `@insert-newline`
-* `@escape-press`
-* `@focus-in`
-* `@focus-out`
-* `@key-down`
-* `@key-press` ([Deprecated Web API](https://developer.mozilla.org/en-US/docs/Web/API/Document/keypress_event))
-* `@key-up`
+Due to legacy code, it is possible to call an action by passing an event argument.
 
+```handlebars
+<label for="input-name">Name:</label>
+<Input
+  @id="input-name"
+  @value={{this.name}}
+  @input={{this.validateName}}
+/>
+```
 
-More [event types](https://api.emberjs.com/ember/release/classes/Component#event-handler-methods) are also supported but these events need to be written in camelCase format, such `mouseEnter`. Note, there are events of the same type in both the list above and linked. Event names listed above must be dasherized. Additional work is performed on these events.
+The argument name is always dasherized (e.g. `@focus-out`, `@input`, `@key-down`). To minimize confusion, we recommend that you use the `{{on}}` modifier. ([Learn more about the `{{on}}` modifier.](../../upgrading/current-edition/action-on-and-fn/#toc_the-on-modifier))
 
-### Checkboxes
+Lastly, Ember Classic provided custom input events `@enter` and `@escape-press`. These are not a part of the HTML standard, but may facilitate accessibility.
 
-You can also use the
-[`<Input />`](https://api.emberjs.com/ember/release/classes/Ember.Templates.components/methods/Input?anchor=Input)
-component to create a checkbox by setting its `@type`:
+Starting with Ember Octane, please consider [writing a modifier](../../upgrading/current-edition/glimmer-components/#toc_writing-your-own-modifiers) to separate concerns: The component manages the state, while the modifier manages interactions with the DOM. If you prefer installing an addon, we recommend the `{{on-key}}` modifier from [ember-keyboard](https://github.com/adopted-ember-addons/ember-keyboard). Your action will receive an actual `event` object.
```

Comment:
```suggestion
The modern, Octane-style way to handle keyboard events is to [write a modifier](../../upgrading/current-edition/glimmer-components/#toc_writing-your-own-modifiers) to separate concerns: The component manages the state, while the modifier manages interactions with the DOM. Your action will receive an actual `event` object. There are also [community-made addons](https://emberobserver.com/?query=keyboard) to help manage keyboard events.

```

Reply:
Saying that an addon is recommended is sticky business. It's preferred to link to Ember Observer unless there is only one option to talk about (i.e. there is no equivalent to ember-concurrency)

Reply:
To my understanding, `ember-keyboard` is the only addon that handles complexities (e.g. `key` vs `code`) in listening to keyboard events and provides a modifier as solution. I don't think it'd be a good idea to ask a developer to create their own modifier and risk not addressing the complexities, then breaking the web.

Since the repo is maintained by `adopted-ember-addons` organization, I was thinking it should be okay for us to recommend this addon? The code example of migrating `@enter` and `@escape-press` can be simple because it assumed that the addon has been installed.

Reply:
I think we could also consider the option of linking to both Ember Observer (in case there will be more keyboard-related addons in the future) and `ember-keyboard`.

```suggestion
The modern, Octane-style way to handle keyboard events is to [write a modifier](../../upgrading/current-edition/glimmer-components/#toc_writing-your-own-modifiers) to separate concerns: The component manages the state, while the modifier manages interactions with the DOM. Your action will receive an actual `event` object.

There are also [community-made addons](https://emberobserver.com/?query=keyboard) to help manage keyboard events. For example, with [ember-keyboard](https://github.com/adopted-ember-addons/ember-keyboard), you can write,
```